### ISSUE TITLE
[bug]: ImportError: urllib3 v2.0 only supports OpenSSL 1.1.1+, currently the 'ssl' module is compiled with LibreSSL 2.8.3. See: https://github.com/urllib3/urllib3/issues/2168

### ISSUE THREAD

**Issue by Dakuan** [invoke-ai/InvokeAI#3358]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### OS

macOS

### GPU

cuda

### What version did you experience this issue on?

latest

### What happened?

attempted install, failed:

```
Installing collected packages: urllib3, numpy, requests
  Attempting uninstall: urllib3
    Found existing installation: urllib3 2.0.2
    Uninstalling urllib3-2.0.2:
      Successfully uninstalled urllib3-2.0.2
  Attempting uninstall: numpy
    Found existing installation: numpy 1.24.3
    Uninstalling numpy-1.24.3:
      Successfully uninstalled numpy-1.24.3
  Attempting uninstall: requests
    Found existing installation: requests 2.30.0
    Uninstalling requests-2.30.0:
      Successfully uninstalled requests-2.30.0
Successfully installed numpy-1.23.5 requests-2.28.2 urllib3-1.26.15
Traceback (most recent call last):
  File "/Users/dominicbarker/Downloads/InvokeAI-Installer/./lib/main.py", line 50, in <module>
    inst.install(**args.__dict__)
  File "/Users/dominicbarker/Downloads/InvokeAI-Installer/lib/installer.py", line 189, in install
    self.instance.configure()
  File "/Users/dominicbarker/Downloads/InvokeAI-Installer/lib/installer.py", line 344, in configure
    import requests  # to catch download exceptions
  File "/private/var/folders/6r/dstg2nwd4_zdl6xldn4k4k0r0000gn/T/invokeai-installer-tmp0qvmyfks/lib/python3.9/site-packages/requests/__init__.py", line 43, in <module>
    import urllib3
  File "/private/var/folders/6r/dstg2nwd4_zdl6xldn4k4k0r0000gn/T/invokeai-installer-tmp0qvmyfks/lib/python3.9/site-packages/urllib3/__init__.py", line 38, in <module>
    raise ImportError(
ImportError: urllib3 v2.0 only supports OpenSSL 1.1.1+, currently the 'ssl' module is compiled with LibreSSL 2.8.3```

### Screenshots

_No response_

### Additional context

_No response_

### Contact Details

_No response_

**Comment by Dakuan** [comment]
I fixed this by upgrading OSX from 13.0 -> 13.1

**Comment by xuhongfei** [comment]
My OSX version is 13.3.1(a). And I have the same issue. Does anyone have approach to resolve this?

**Comment by Akanta21** [comment]
Not sure if this would work for you.

I resolved it using this method in the interim by lowering the urllib3 requirement:

pip install urllib3==1.26.6

**Comment by gdjohn4s** [comment]
@Akanta21 I resolved this issue in the same way.
Thanks!

**Comment by xuhongfei** [comment]
> My OSX version is 13.3.1(a). And I have the same issue. Does anyone have approach to resolve this?

Thanks @Akanta21 . This worked for me:
pip install urllib3==1.26.6

**Comment by robotica-labs** [comment]
Same error 13.3.1 (a) (22E772610a) I will try lowering urllib3

**Comment by billygrahamram** [comment]
Once I do 
`pip install urllib3==1.26.6`

I get this error

`ImportError: cannot import name 'get_host' from 'urllib3.util.url'`


I tested the code in the Ubuntu system and it works perfectly. The problem is in the Macbook.

**Comment by mr0303** [comment]
pip uninstall urllib3
pip install 'urllib3<2.0
the above worked for me

**Comment by ajay2929** [comment]
Thanks. I did resolve using this same.

**Comment by billygrahamram** [comment]
> pip uninstall urllib3 pip install 'urllib3<2.0' the above worked for me

Thanks mr0303. This worked!

**Comment by jeremyklein** [comment]
I downgraded urllib3 and I am still getting the same behavior. ImportError: cannot import name 'get_host' from 'urllib3.util.url'

 The output of pip freeze from my virtual env is:

aiofiles==22.1.0
aiohttp==3.8.4
aiosignal==1.3.1
aiosqlite==0.19.0
anyio==3.6.2
appnope==0.1.3
argon2-cffi==21.3.0
argon2-cffi-bindings==21.2.0
arrow==1.2.3
asttokens==2.2.1
async-timeout==4.0.2
attrs==23.1.0
Babel==2.12.1
backcall==0.2.0
beautifulsoup4==4.12.2
bleach==6.0.0
certifi==2023.5.7
cffi==1.15.1
charset-normalizer==3.1.0
comm==0.1.3
dataclasses-json==0.5.7
debugpy==1.6.7
decorator==5.1.1
defusedxml==0.7.1
executing==1.2.0
fastjsonschema==2.16.3
fqdn==1.5.1
frozenlist==1.3.3
greenlet==2.0.2
idna==3.4
importlib-metadata==6.6.0
importlib-resources==5.12.0
ipykernel==6.23.0
ipython==8.12.2
ipython-genutils==0.2.0
isoduration==20.11.0
jedi==0.18.2
Jinja2==3.1.2
json5==0.9.11
jsonpointer==2.3
jsonschema==4.17.3
jupyter-events==0.6.3
jupyter-ydoc==0.2.4
jupyter_client==8.2.0
jupyter_core==5.3.0
jupyter_server==2.5.0
jupyter_server_fileid==0.9.0
jupyter_server_terminals==0.4.4
jupyter_server_ydoc==0.8.0
jupyterlab==3.6.3
jupyterlab-pygments==0.2.2
jupyterlab_server==2.22.1
langchain==0.0.165
MarkupSafe==2.1.2
marshmallow==3.19.0
marshmallow-enum==1.5.1
matplotlib-inline==0.1.6
mistune==2.0.5
multidict==6.0.4
mypy-extensions==1.0.0
nbclassic==1.0.0
nbclient==0.7.4
nbconvert==7.4.0
nbformat==5.8.0
nest-asyncio==1.5.6
notebook==6.5.4
notebook_shim==0.2.3
numexpr==2.8.4
numpy==1.24.3
openai==0.27.6
openapi-schema-pydantic==1.2.4
packaging==23.1
pandocfilters==1.5.0
parso==0.8.3
pexpect==4.8.0
pickleshare==0.7.5
pkgutil_resolve_name==1.3.10
platformdirs==3.5.0
prometheus-client==0.16.0
prompt-toolkit==3.0.38
psutil==5.9.5
ptyprocess==0.7.0
pure-eval==0.2.2
pycparser==2.21
pydantic==1.10.7
Pygments==2.15.1
pyrsistent==0.19.3
python-dateutil==2.8.2
python-json-logger==2.0.7
pytz==2023.3
PyYAML==6.0
pyzmq==25.0.2
requests==2.30.0
rfc3339-validator==0.1.4
rfc3986-validator==0.1.1
Send2Trash==1.8.2
six==1.16.0
sniffio==1.3.0
soupsieve==2.4.1
SQLAlchemy==2.0.13
stack-data==0.6.2
tenacity==8.2.2
terminado==0.17.1
tinycss2==1.2.1
tomli==2.0.1
tornado==6.3.1
tqdm==4.65.0
traitlets==5.9.0
typing-inspect==0.8.0
typing_extensions==4.5.0
uri-template==1.2.0
urllib3==1.26.6
wcwidth==0.2.6
webcolors==1.13
webencodings==0.5.1
websocket-client==1.5.1
y-py==0.5.9
yarl==1.9.2
ypy-websocket==0.8.2
zipp==3.15.0

**Comment by Etlay** [comment]
Same problem
OS version: 13.3.1

**Comment by TheMellyBee** [comment]
I'm still having the issue with OS version: 13.3.1

Same change with the ImportError after down grading.

**Comment by Mont9165** [comment]
Same error happened OS version: 13.3.1.
Thanks @mr0303 !!  I resolved this error in the same way.

**Comment by lehigh123** [comment]
I'm still having the issue with OS version: 13.3.1. Tried all solutions above and none worked

**Comment by bcapps1259** [comment]
After struggling to get this running on my new Mac Mini M2 with Ventura 13.4, here is my solution:

1) Start installer, but do not continue. This creates a set of temporary install folders.

2) Navigate to the folder below:
/private/var/folders/3z/tl94bhbd0sl5gzlj29__6jwm0000gn/T/invokeai-installer-tmppc1li4ec/lib/python3.9/site-packages

3) Delete two urllib3 folders.

4) Continue with installation.

It worked for me, and I hope that it works for you too!

**Comment by Cocii** [comment]
> 

THX!!! But for me, the path is "/private/var/folders/ms/l2jgzvsj1xvgz_b45r423kg00000gn/T/invokeai-installer-tmpg98rn5aj/lib/python3.9/site-packages"

the path will appear at the error
"
    import urllib3
  File "/private/var/folders/ms/l2jgzvsj1xvgz_b45r423kg00000gn/T/invokeai-installer-tmpg98rn5aj/lib/python3.9/site-packages/urllib3/__init__.py", line 38, in <module>
    raise ImportError(
ImportError: urllib3 v2.0 only supports OpenSSL 1.1.1+, currently the 'ssl' module is compiled with LibreSSL 2.8.3. See: https://github.com/urllib3/urllib3/issues/2168
"

**Comment by NebularNerd** [comment]
> 

I'm testing a Python script on a crusty GoDaddy hosted server, this did the trick for me as they do not offer 3.1x versions in cPanel yet :(

**Comment by NebularNerd** [comment]
> pip uninstall urllib3
> pip install 'urllib3<2.0'

Thanks @mr0303,  I'm testing a Python script on a crusty GoDaddy hosted server, this did the trick for me as they do not offer 3.1x versions in cPanel yet :(

**Comment by prasaddd77** [comment]
> I fixed this by upgrading OSX from 13.0 -> 13.1

How do I do that ?
